brain/circuit.h

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2013-2015, EPFL/Blue Brain Project
+ *                          Juan Hernando <jhernando@fi.upm.es>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef BRAIN_CIRCUIT
+#define BRAIN_CIRCUIT
+
+#include <brain/api.h>
+#include <brain/types.h>
+
+namespace brain
+{
+
+/** Read access to a circuit database
+ *
+ * This class provides convenience functions to access information about the
+ * cells inside the circuit and their morphologies.
+ */
+class Circuit
+{
+public:
+    /** Coordinate system to use for circuit morphologies */
+    enum Coordinates
+    {
+        COORDINATES_GLOBAL,
+        COORDINATES_LOCAL
+    };
+
+    /**
+     * Opens a circuit for read access.
+     * @param source the URI to the CircuitConfig or BlueConfig file.
+     */
+    BRAIN_API Circuit( const URI& source );
+
+    /**
+     * Opens a circuit for read access.
+     * @param config The object representing the BlueConfig.
+     */
+    BRAIN_API Circuit( const brion::BlueConfig& blueConfig );
+
+    BRAIN_API  ~Circuit();
+
+    /**
+     * @return The set of GIDs for the given target name. If empty it will
+     *         return the circuit target specified on the
+     *         BlueConfig/CircuitConfig file.
+     *         If the target cannot be found or an empty string was given and
+     *         there is no circuit target, the return value is an empty set.
+     */
+    BRAIN_API GIDSet getGIDs( const std::string& target = "" ) const;
+
+    /** @return The set of URIs to access the morphologies of the given cells */
+    BRAIN_API URIs getMorphologyURIs( const GIDSet& gids ) const;
```

Comment:
I thought we had agreed on having this method yesterday.

It's used internally, but I think that it can find uses outside the implementation.
You can use it to get the paths to the morphologies on load the data you're interested in, in the moment you want.